basetest.pm

```diff
@@ -640,24 +640,26 @@ sub parse_serial_output_qemu {
     my $json = myjsonrpc::read_json($autotest::isotovideo);
 
     my $die = 0;
+    my %regexp_matched;
     # loop line by line
     for my $line (split(/^/, $json->{serial})) {
         chomp $line;
         # only two keys allowed
         for my $type (qw(soft hard)) {
-            for my $regexp (@{$failures->{$type}}) {
+            for my $regexp_table (@{$failures->{$type}}) {
                 # If you want to match a simple string please be sure that you create it with quotemeta
-                if ($line =~ /$regexp/) {
+                my $regexp = $regexp_table->{pattern};
+                if (!exists $regexp_matched{$regexp} and $line =~ /$regexp/) {
+                    $regexp_matched{$regexp} = 1;
```

Comment:
I guess this mechanism is required for soft failures only. @dasantiago is it the case?. And advantage of having regexp as a key is that all of them will be unique. So, please, consider using it as a key and put $type as a part of hash. Then we can iterate over keys and values of hash in single loop like `while(my($k, $v) = each %h)`, but that's optional.

Reply:
@rwx788 
My suggestion was for **consecutive** exceptions that are the same, independently of soft / hard.

IMHO, having 30 consecutive exceptions that are the same, only creates noise.

This doesn't address the consecutive part ;-)